### PR TITLE
Docs: Add code snippets for Web Component

### PR DESCRIPTION
**Pull request by gaetanmaisse**
Issue: #14120 (and #9693?)

## What I did

I added Web Component code snippets for `Get Started` and `Writing Stories` sections.
I covered almost all the snippets, the remaining ones are related to things I'm not sure they can be achieved with `@storybook/web-components`. I didn't cover the other sections, like addons for instance, for the same reason.

## How to test

- Read the files to check I didn't make any typo or obvious mistakes and trust me for the story formats :-p

### CODE REVIEW

**Review by shilman — APPROVED** [review]
Excellent @gaetanmaisse -- still lots more to do here but it's a great start. So excited to see this getting filled out... 🔥 

@jonniebigodes can you review?

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@gaetanmaisse this is awesome. Can't thank you enough for putting this together and get the initial set of snippets for web components into the documentation. Left one item that needs to be addressed and this should be good to go.

File: docs/snippets/web-components/your-component.js.mdx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+// This default export determines where your story goes in the story list
+export default {
+  title: 'YourComponent',
+};
+
+const Template = (
+  /* the args you need here will depend on your component */
+  { aProperty }
+) => html`<your-component .yourProperty=${aProperty}></your-component>`;
```
[the remark this answers is not in the document]
Comment (gaetanmaisse) [reply docs/snippets/web-components/your-component.js.mdx:16]: @jonniebigodes should I add something in the `.args` too? To understand that the things defined in args are the arguments of the template function. Like: 
```js
// 👇 We create a “template” of how args map to rendering
const Template = ({ aProperty }) => html`<your-component .yourProperty=${aProperty}></your-component>`;

export const FirstStory = Template.bind({});

FirstStory.args = {
  /* 👇 The args you need here will depend on your component */
  aProperty: 'a value'
};
```

[the remark this answers is not in the document]
Comment (jonniebigodes) [reply docs/snippets/web-components/your-component.js.mdx:16]: Yes. Indeed you do. I forgot to update the comment to address that. Sorry about that.

[the remark this answers is not in the document]
Comment (gaetanmaisse) [reply docs/snippets/web-components/your-component.js.mdx]: Done @jonniebigodes I just updated my commit :)

**Comment by ariefurtado** [comment]
@gaetanmaisse your web-components snippets really saved my day!

As you mentioned you covered *almost all the snippets*, but i really need a way to do **Stories for two or more components**.
Here goes a simple example of the result i'm looking for:

```html
<my-component>
     <my-inner-component></my-inner-component>
     <my-inner-component></my-inner-component>
     <my-inner-component></my-inner-component>
</my-component>
```

I have followed all the tips about the web-component implementation with lit-html and I've tried many ways to achieve this goal, but unfortunately I could not come up with an approach that works.
As we can see on this snippet from the docs, when using SB with React it is as simple as import and use:

```tsx
import React from 'react';

//👇 We're importing the necessary stories from ListItem
import { Selected, Unselected } from './ListItem.stories';

export const ManyItems = (args) => (
  <List {...args}>
    <Selected {...Selected.args} />
    <Unselected {...Unselected.args} />
    <Unselected {...Unselected.args} />
  </List>
);
```

But for some reason I can't find any examples on this for web-components.

**Does anyone have an ideia on how to implement that?**

**Comment by gaetanmaisse** [comment]
@ariefurtado I played a bit more with the snippets and may have found what would be the Web Components syntax for ⬆️ 
Does the following do what you're expecting?
```js
import { html } from 'lit-html';

export const ManyItems = (args) => {
  return html`
    <div>
      <div>${Regular({ ...args, ...Regular.args })}</div>
      <div>${Selected({ ...args, ...Selected.args })}</div>
      <div>${Unselected({ ...args, ...Unselected.args })}</div>
    </div>
  `;
}
```

**Comment by ariefurtado** [comment]
Hello @gaetanmaisse,

Thank you for the feedback!

I've played a little more and it seems to work perfectly with your instructions.
I beleve it should be added to the snippets in the docs section **Stories for two or more components**.

image

**Comment by jonniebigodes** [comment]
@ariefurtado indeed. I'm extremely grateful to @gaetanmaisse for his awesome job on getting an initial set of web-components snippets into the documentation. If you're ok I would really appreciate your help as well with this and I can follow up with you. So that we continue to expand the available snippets. Let me know what you think and we'll go from there. Sounds good?

**Comment by ariefurtado** [comment]
Hi @jonniebigodes,

I don't know how much I can help, but count on me, I'll keep an eye on this thread.

I've been implementing a pipeline for the front-end dev team at work. We want to have a *per mono repo* web-components project (stenciljs + storybook).

The interesting part is that we'll have an independant web-component lib (also in stenciljs + storybook). The *mono repo* web-components project will have an independant web-component lib (also in stenciljs + storybook) as dependency to use for component composition. As the components matures and turn more to a generic/agnostic component, it will be migrated to the external lib. Our front-end team will be able to stay *focused on value* and scale to agnostic.

image


I would appreciate if someone have any thoughts on this, good or bad ;)

--

Now @jonniebigodes, It seems to me that I'll be working with web-components and storybook for a long time, then, let me know if you need a hand and thanks on advance for all the feedbacks.

**Pull request merged**